### ISSUE TITLE
learn exe don't use all training data even without rejet and loop 1

### ISSUE THREAD

**Issue by deedstacktyk** [nodchip/Stockfish#70]
Hi Nodchip,

I don't find the good settings for eval_save_interval and loss_output_interval to use all the generated sfens at training phase.

Some examples with 1b sfens in a training .bin file :

- learn "256x2" loop 1 + eval_save_interval 250m + loss_output_interval 50m => writes several nn.bin in evalsave subdirectories until 700m sfens then continues to calculate until 950m then i see the "end of file" mention and neither LOSS value nor new nn.bin in evalsave subdirectories or evalsave/final

- learn "384x2" loop 1 + eval_save_interval 200m + loss_output_interval 50m => writes several nn.bin in evalsave subdirectories until 750m sfens then continues to calculate until 950m then i see the "end of file" mention and neither LOSS value nor new nn.bin in evalsave subdirectories or evalsave/final

No help at discord channel...

**Comment by nodchip** [comment]
Could you please set "100" to "loop"?  It specifies how many times the trainer read the training data.

**Comment by deedstacktyk** [comment]
Do you mean i need a loop greater than 1 to use 100% of the training data !?

Here it takes 13 hours to reach 950m with loop 1. I'm afraid If i set loop 100 (100 * 13 hrs = about 54 days !?), it may takes a very looong times to solve this issue.

**Comment by nodchip** [comment]
> Do you mean i need a loop greater than 1 to use 100% of the training data !?
Yes.

> Here it takes 13 hours to reach 950m with loop 1. I'm afraid If i set loop 100 (100 * 13 hrs = about 54 days !?), it may takes a very looong times to solve this issue.
A training will automatically finish, once the loss is converged even though the loop is set to 100.

**Comment by deedstacktyk** [comment]
OK thank you very much !

So i will retest with loop 2 because i think reusing several times the same data was not optimal at learning phase...

**Comment by FireFather** [comment]
@deedstacktyk I believe he is recommending using
' loop 100 '

**Comment by deedstacktyk** [comment]
why "loop 1" don't use 100% of the training data ? why should i start for multiple loops in order to train only one all the data ?

**Comment by nodchip** [comment]
I think that it is an implementation problem.  There are no plans to fix it because it is not a critical issue.  Pull requests to fix are welcome.

I'm sorry to say this one but here is not a right place to ask questions.  This is a place to report issues.  I recommend to ask further questions in the #help channel in the unofficial Stockfish Discord server.

I will close this issue.
